hail/src/main/scala/is/hail/expr/ir/agg/LinearRegressionAggregator.scala

```diff
@@ -13,8 +13,8 @@ object LinearRegressionAggregator extends StagedAggregator {
   val vector = PArray(PFloat64(true), true)
   val scalar = PFloat64(true)
   val stateType: PTuple = PTuple(true, vector, vector, PInt32(true))
-  val nrVec = PArray(PFloat64())
-  def resultType = PStruct("xty" -> nrVec, "beta" -> nrVec, "diag_inv" -> nrVec, "beta0" -> nrVec)
+  val nrVec = PCanonicalArray(PFloat64(true), true)
```

Comment:
I think you can collapse `nrVec` and `vector`, right? I forget why I needed the non-required vector type, I'm guessing to get around requiredness bugs.

Reply:
oh, ha -- nrVec == non-required vector...

Updated.